labgrid/remote/client.py

```diff
@@ -189,14 +196,41 @@ def onChallenge(self, challenge):
                             resource_name, exporter, group_name, resource.cls, resource_name))
                         print(indent(pformat(resource.asdict()), prefix="      "))
         else:
+            results = []
             for exporter, groups in sorted(filtered.items()):
                 for group_name, group in sorted(groups.items()):
                     for resource_name, resource in sorted(group.items()):
-                        print("{}/{}/{}".format(exporter, group_name, resource.cls))
+                        resource_path = (exporter, group_name, resource.cls, resource_name)
+                        places = list(self._get_places_by_resource(resource_path))
+                        # most recently changed place first
+                        places = sorted(places, key=lambda p: p.changed, reverse=True)
```

Comment:
The places computation and sort are done unconditionally and are not used in the old code path.
Please move the computation so the previous old code path remains at the same speed as before.

Reply:
Or just move it into a sperate command?

Reply:
Does this really have an effect on performance? If not, should we always show the matching places?

Reply:
I did not touch the output of `labgrid-client resources` to keep it stable (e.g. for scripting purposes, e.g. tab completion).

Regarding performance: for each resource all places' getmatch() is called. This uses fnmatchcase() up to 4 times (exporter, group, class, name). fnmatchcase does caching, but [only the 256 most recent patterns](https://github.com/python/cpython/blob/3.6/Lib/fnmatch.py#L38). So yes, I guess there are differences in performance, especially for larger setups.

Reply:
Place computation is now done conditionally.